### PR TITLE
Move processing of withdrawals seperate of Withdrawals

### PR DESCRIPTION
**Pull request by greenbigfrog**

### CODE REVIEW

File: schema.sql [hunk 1]
```diff
@@ -38,6 +38,16 @@ CREATE TABLE coin_transactions (
        txhash text PRIMARY KEY,
        status text NOT NULL,
 
+       created_time timestamptz NOT NULL DEFAULT now()
+);
+
+CREATE TABLE withdrawals (
+       id serial PRIMARY KEY,
+       status text DEFAULT 'pending',
```
Comment (z64) [inline schema.sql]: [enums](https://www.postgresql.org/docs/9.6/static/datatype-enum.html) are cool

File: src/discordtipbot/discordbot.cr [hunk 2]
```diff
@@ -6,6 +6,9 @@ class DiscordBot
   USER_REGEX = /<@!?(?<id>\d+)>/
   START_TIME = Time.now
   TERMS      = "In no event shall this bot or its dev be responsible for any loss, theft or misdirection of funds."
+  LOADING    = "<a:loading:412291714192113667>"
+  CURSOR     = "<a:cursor:412292067642048512>"
+  CHECK      = "<:check:412291051290886155>"
```
Comment (z64) [inline src/discordtipbot/discordbot.cr]: I would kind of like to see a namespace for constants;
```cr
module Emoji
  Loading = "<a:loading:412291714192113667>"
  Cursor  = "<a:cursor:412292067642048512>"
  Check   = "<:check:412291051290886155>"
end
```

File: src/discordtipbot/discordbot.cr [hunk 3]
```diff
@@ -374,7 +391,7 @@ class DiscordBot
     when false
       reply(msg, "**ERROR**: There was a problem trying to withdraw. Please try again later. If the problem persists, please contact the dev for help in #{@config.prefix}support")
     when true
-      reply(msg, "Successfully withdrew **#{amount} #{@config.coinname_short}** to **#{address}**")
+      reply(msg, "Pending withdrawal of **#{amount} #{@config.coinname_short}** to **#{address}**. *Processing shortly*" + CURSOR)
```
Comment (z64) [inline src/discordtipbot/discordbot.cr]: I kind of think "processing shortly" should be removed.

File: src/discordtipbot/tipbot.cr [hunk 4]
```diff
@@ -70,6 +66,25 @@ class TipBot
     true
   end
 
+  def process_pending_withdrawals
+    tuple = {id: Int32, from_id: Int64, address: String, amount: BigDecimal}
```
Comment (z64) [inline src/discordtipbot/tipbot.cr]: I might recommend a `record` instead.

File: src/discordtipbot/discordbot.cr (hunk 3 above)
Comment (greenbigfrog) [reply src/discordtipbot/discordbot.cr]: Like completely get rid of it? Not sure if people will really understand what's happening

File: src/discordtipbot/tipbot.cr (hunk 4 above)
Comment (greenbigfrog) [reply src/discordtipbot/tipbot.cr]: Mind elaborating? calling what I'm calling `tuple` `record`?

Comment (greenbigfrog) [reply src/discordtipbot/tipbot.cr]: After thinking about it, I think that's what you meant :smiley:

**Pull request merged**